sw/device/lib/dif/dif_hmac.c

```diff
@@ -0,0 +1,400 @@
+    case kDifHmacEndiannessBig:
+      device_config |= (1 << HMAC_CFG_DIGEST_SWAP);
+      break;
+    case kDifHmacEndiannessLittle:
+      break;
+    default:
+      return kDifHmacError;
+  }
+
+  // Write the configuration.
+  mmio_region_write32(config->base_addr, HMAC_CFG_REG_OFFSET, device_config);
+
+  return kDifHmacOk;
+}
+
+dif_hmac_result_t dif_hmac_irq_state_get(const dif_hmac_t *hmac,
+                                         dif_hmac_interrupt_t irq_type,
+                                         dif_hmac_enable_t *state) {
+  if (hmac == NULL || state == NULL) {
+    return kDifHmacBadArg;
+  }
+
+  ptrdiff_t offset;
+  if (!irq_bit_offset_get(irq_type, &offset)) {
+    return kDifHmacError;
+  }
+
+  // Get the interrupt state.
+  bool enabled = mmio_region_get_bit32(hmac->base_addr,
+                                       HMAC_INTR_STATE_REG_OFFSET, offset);
+  *state = (enabled ? kDifHmacEnable : kDifHmacDisable);
+
+  return kDifHmacOk;
+}
+
+dif_hmac_result_t dif_hmac_irq_state_clear(const dif_hmac_t *hmac,
+                                           dif_hmac_interrupt_t irq_type) {
+  if (hmac == NULL) {
+    return kDifHmacBadArg;
+  }
+
+  ptrdiff_t offset;
+  if (!irq_bit_offset_get(irq_type, &offset)) {
+    return kDifHmacError;
+  }
+
+  // Clear the interrupt state.
+  mmio_region_write_only_set_bit32(hmac->base_addr, HMAC_INTR_STATE_REG_OFFSET,
+                                   offset);
+
+  return kDifHmacOk;
+}
+
+dif_hmac_result_t dif_hmac_irqs_disable(const dif_hmac_t *hmac,
+                                        uint32_t *state) {
+  if (hmac == NULL) {
+    return kDifHmacBadArg;
+  }
+
+  // Pass the interrupt state back to the caller.
+  if (state != NULL) {
+    *state = mmio_region_read32(hmac->base_addr, HMAC_INTR_STATE_REG_OFFSET);
+  }
+
+  // Disable all interrupts.
+  mmio_region_write32(hmac->base_addr, HMAC_INTR_STATE_REG_OFFSET, 0u);
+
+  return kDifHmacOk;
+}
+
+dif_hmac_result_t dif_hmac_irqs_restore(const dif_hmac_t *hmac,
+                                        uint32_t state) {
+  if (hmac == NULL) {
+    return kDifHmacBadArg;
+  }
+
+  // Restore interrupt state.
+  mmio_region_write32(hmac->base_addr, HMAC_INTR_STATE_REG_OFFSET, state);
+
+  return kDifHmacOk;
+}
+
+dif_hmac_result_t dif_hmac_irq_control(const dif_hmac_t *hmac,
+                                       dif_hmac_interrupt_t irq_type,
+                                       dif_hmac_enable_t enable) {
+  if (hmac == NULL) {
+    return kDifHmacBadArg;
+  }
+
+  ptrdiff_t offset;
+  if (!irq_bit_offset_get(irq_type, &offset)) {
+    return kDifHmacError;
+  }
+
+  // Enable/Disable interrupt state.
+  if (enable == kDifHmacEnable) {
+    mmio_region_nonatomic_set_bit32(hmac->base_addr, HMAC_INTR_STATE_REG_OFFSET,
+                                    offset);
+  } else {
+    mmio_region_nonatomic_clear_bit32(hmac->base_addr,
+                                      HMAC_INTR_STATE_REG_OFFSET, offset);
+  }
```

Comment:
Wrong register, should write to `HMAC_INTR_ENABLE_REG_OFFSET`. Also, please consider dropping "state" from the comment.